torch/utils/data/_utils/worker.py

```diff
@@ -220,9 +221,22 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
         signal_handling._set_worker_signal_handlers()
 
         torch.set_num_threads(1)
+
+        # Note [RNG re-seeding in Dataloader workers]
+        # We need to explicitly re-seed all RNGs here (builtin random, numpy, torch) to ensure that the RNGs are
+        # different across workers.
+        # We have to seed the default RNG and the user-made ones differently otherwise the RNG of all Generators
+        # would be the same as the default one.
+        # For each Generator we have to draw a seed that depends on its current state (done in _generate_seed()),
+        # otherwise all Generator instances within a given worker would have the same RNG.
         seed = base_seed + worker_id
         random.seed(seed)
-        torch.manual_seed(seed)
+        generators = [o for o in gc.get_objects() if isinstance(o, torch.Generator)]
+        for generator in generators:
+            if generator is torch.random.default_generator:
+                torch.manual_seed(seed)
+            elif generator.device.type != "cuda":
```

Comment:
Gotta add a comment saying why not to reinitialize CUDA generators. Actually, would it be safer to ONLY reseed cpu generator instead? Saves you from bug reports about xpu, etc.